src/views/viewer/ChannelView/ChannelView.tsx

```diff
@@ -192,9 +259,60 @@ export const ChannelView: React.FC = () => {
 
 const usePagination = (currentTab: number) => {
   const [currentPage, setCurrentPage] = useState(0)
+  const [currentSearchPage, setCurrentSearchPage] = useState(0)
   // reset the pagination when changing tabs
   useEffect(() => {
     setCurrentPage(0)
+    setCurrentSearchPage(0)
   }, [currentTab])
-  return { currentPage, setCurrentPage }
+  return { currentPage, setCurrentPage, currentSearchPage, setCurrentSearchPage }
+}
+
+type useSearchVideosParams = {
+  id: string
+}
+const useSearchVideos = ({ id }: useSearchVideosParams) => {
+  const [isSearchInputOpen, setIsSearchingInputOpen] = useState(false)
+  const [searchQuery, setSearchQuery] = useState('')
+  const [triggeredSearchQuery, setTriggeredSearchQuery] = useState('')
+  const [isSearching, setIsSearching] = useState(false)
+  const searchInputRef = useRef<HTMLInputElement>(null)
+  // TODO: bug this is not querying by channel Id and results don't seem accurate
+  const { data: searchData, loading: loadingSearch, error: errorSearch } = useSearch({
+    text: triggeredSearchQuery,
+    whereVideo: {
+      isPublic_eq: true,
+      mediaAvailability_eq: AssetAvailability.Accepted,
+      thumbnailPhotoAvailability_eq: AssetAvailability.Accepted,
+    },
+    whereChannel: {
+      id_in: [id],
```

Comment:
you should update `extendedQueryNode` and add `id_eq` field. Maybe it will fix the issue